src/org/openstreetmap/josm/actions/search/SearchCompiler.java

```diff
@@ -1554,6 +1558,44 @@ public String toString() {
         }
     }
 
+    /**
+     * Matches presets.
+     */
+    private static class Preset extends Match {
+        private List<TaggingPreset> ps;
+
+        Preset(String presetName) throws ParseError {
+
+            if (presetName == null)
+                throw new ParseError("Preset name cannot be null");
+
+            Collection<TaggingPreset> ts = TaggingPresets.getTaggingPresets();
+            ps = new ArrayList<>();
```

Comment:
You can use streams for this:
```java
ts.stream().filter(preset -> presetName.equalsIgnoreCase(preset.getSimpleName())).collect(Collectors.toList())
```

By using equals that way, you don't need the null check

Please use longer variable names.